File: server/src/main/java/pl/sherpout/server/api/user/dto/UserDto.java
package pl.sherpout.server.api.user.dto;

import lombok.Getter;
import lombok.Setter;
import pl.sherpout.server.config.security.group.UserGroup;

@Getter
@Setter
public class UserDto {
    private UserGroup group;
    private String firstName;
    private String lastName;
    private String email;
    private boolean emailVerified;
}

File: sherpout_mobile/lib/common/dto/user_dto.dart
import 'package:sherpoutmobile/common/dto/user_group.dart';

class UserDto {
  String id;
  UserGroup group;
  String firstName;
  String lastName;
  String email;
  bool emailVerified;

  UserDto({
    required this.id,
    required this.group,
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.emailVerified,
  });

  factory UserDto.fromJson(Map<String, dynamic> json) {
    return UserDto(
      id: json['id'] as String,
      group: UserGroup.values.firstWhere(
            (e) => e.name.toUpperCase() == json['group'],
        orElse: () => throw ArgumentError('Invalid UserGroup: ${json['group']}'),
      ),
      firstName: json['firstName'] as String,
      lastName: json['lastName'] as String,
      email: json['email'] as String,
      emailVerified: json['emailVerified'] as bool,
    );
  }
}